game_api.txt

```diff
@@ -1115,3 +1115,33 @@ This function registers a shapeless recipe that takes `ingredient`
 and `result` as input and outputs `result`.
 
 The metadata of the input `result` is copied to the output `result`.
+
+
+Log API
+-------
+
+Logs action of the player with a node at a certain position.
+By default only actions of real players are logged.
+Actions of non-players (usually machines) are logged only when
+setting `log_non_player_actions` is enabled.
+A player is considered non-player if `player:is_player()` returns
+`false` or `player.is_fake_player` is truthy. The use of
+`is_fake_player` is an unofficial standard between mods.
+These non-players are marked by the content of `is_fake_player`
+(if it is a string) or a "*" in brackets after the player name in
+the log.
+
+`default.log_player_action(player, ...)`
+
+ * `player`        The player who performed the action
+ * `message_parts` Any mumber of message parts describing the action
+                   in 3rd person singular present tense. It can also
+		   contain a `vector` which is logged as " at (X,Y,Z)"
```

Comment:
doing this is inflexible and error-prone

you even made a mistake in this PR:
<pre>
default.log_player_action(placer,
  "tried to place", itemstack:get_name(),
  "at protected position", pos_under)
</pre>
expands to "foo tried to place mod:item at protected position **at** (1,2,3)"

Reply:
not really a mistake (see https://github.com/minetest/minetest_game/pull/2941#issuecomment-1105221099 & https://github.com/minetest/minetest_game/pull/2941#issuecomment-1106347160) but fixed in https://github.com/minetest/minetest_game/pull/2941/commits/589bc4e5a4e34014efed7604d3f21e4fb862764f  :-)

Another possible error was a message like:
```lua
minetest.log("action", player_name .. " dies at " .. pos_string .. ". Inventory dropped")
--> "PLAYERNAME dies at (1,2,4). Inventory dropped"
```
vs.
```lua
default.log_player_action(player, "dies at ", pos, ". Inventory dropped")
--> "PLAYERNAME dies at (1,2,4) . Inventory dropped"
```
fixed in https://github.com/minetest/minetest_game/pull/2941/commits/238a93fcb353627b507315960b36a6152a7e93e9
```lua
default.log_player_action(player, "dies at ", pos, ". Inventory dropped")
--> "PLAYERNAME dies at (1,2,4). Inventory dropped"
```